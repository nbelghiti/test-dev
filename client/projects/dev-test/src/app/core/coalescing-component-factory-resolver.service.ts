/* tslint:disable */
/**
 * See https://github.com/jonrimmer/angular-coalescing-component-factory-resolver/blob/master/src/app/coalescing-component-factory-resolver.service.ts
 * Discussion & bug ticket: https://github.com/angular/angular/issues/14324#issuecomment-481898762
 *
 * Entry components are only available within a lazy loaded module, breaking dynamic generation (e.g. dialogs). This monkey patch enables resolving entry components from lazy loaded modules as well.
 *
 *
 * Provide in root module:
 *
 * @NgModule({
 *   providers: [CoalescingComponentFactoryResolver]
 * })
 * class AppModule {
 *   constructor(coalescingResolver: CoalescingComponentFactoryResolver) {
 *     coalescingResolver.init();
 *   }
 * }
 *
 *
 * Use in lazy loaded module:
 *
 *
 * @NgModule({})
 * export class LazyModule {
 *   constructor(
 *     coalescingResolver: CoalescingComponentFactoryResolver,
 *     localResolver: ComponentFactoryResolver
 *   ) {
 *     coalescingResolver.registerResolver(localResolver);
 *   }
 * }
 *
 *
 * How it works: When initialised, it injects the root app's ComponentFactoryResolver and monkey patches the resolveComponentFactory method
 * to call its own implementation. This implementation first tries resolving the component factory on all the registered lazy-module resolvers,
 * then falls back to the root app resolver (there's a bit of extra logic to avoid cyclic calls, but that's the gist).
 */
/* tslint:enable */

import {
	ComponentFactory,
	ComponentFactoryResolver,
	Injectable,
	Type
} from '@angular/core';

@Injectable()
export class CoalescingComponentFactoryResolver extends ComponentFactoryResolver {
	private rootResolve: (component: Type<any>) => ComponentFactory<any>;

	private inCall = false;

	private readonly resolvers = new Map<
		ComponentFactoryResolver,
		(component: Type<any>) => ComponentFactory<any>
	>();

	constructor(private readonly rootResolver: ComponentFactoryResolver) {
		super();
	}

	init() {
		this.rootResolve = this.rootResolver.resolveComponentFactory;
		this.rootResolver.resolveComponentFactory = this.resolveComponentFactory;
	}

	public registerResolver(resolver: ComponentFactoryResolver) {
		const original = resolver.resolveComponentFactory;
		this.resolvers.set(resolver, original);
	}

	public resolveComponentFactory = <T>(
		component: Type<T>
	): ComponentFactory<T> => {
		// Prevents cyclic calls.
		if (this.inCall) {
			return null;
		}

		this.inCall = true;
		try {
			const result = this.resolveInternal(component);
			return result;
		} finally {
			this.inCall = false;
		}
	}

	private resolveInternal = <T>(component: Type<T>): ComponentFactory<T> => {
		for (const [resolver, fn] of Array.from(this.resolvers.entries())) {
			try {
				const factory = fn.call(resolver, component);
				if (factory) {
					return factory;
				}
			} catch { }
		}

		return this.rootResolve.call(this.rootResolver, component);
	}
}
