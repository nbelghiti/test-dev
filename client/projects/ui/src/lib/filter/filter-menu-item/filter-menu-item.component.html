<a
	[routerLink]="link"
	*ngIf="link; else buttonTpl"
	class="c-filter-menu-item"
	[ngClass]="{
		'c-filter-menu-item--active': active
	}"
	[title]="label"
	[queryParams]="queryParams"
>
	<span class="c-filter-menu-item__value">{{ label }}</span>

	<div class="c-filter-menu-item__icon">
		<ui-icon name="check"></ui-icon>
	</div>
</a>


<ng-template #buttonTpl>
	<button
		(click)="clickHandler()"
		class="c-filter-menu-item"
		[ngClass]="{
			'c-filter-menu-item--active': active
		}"
		[title]="label"
	>
		<span class="c-filter-menu-item__value">{{ label }}</span>

		<div class="c-filter-menu-item__icon">
			<ui-icon name="check"></ui-icon>
		</div>
	</button>
</ng-template>
