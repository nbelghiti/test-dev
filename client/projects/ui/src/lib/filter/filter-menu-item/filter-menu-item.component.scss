// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: List item
// =============================================================================
.c-filter-menu-item {
	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	padding: 0.4rem 1.5rem;

	text-decoration: none;

	background: none;
	border: none;

	transition: $animation-base;

	cursor: pointer;

	&:hover {
		background-color: $ui-color-success-light;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-filter-menu-item__value {
	overflow: hidden;
	margin-right: 1.6rem;

	font-family: $font-family-secondary;
	font-size: $font-size-base;
	line-height: $line-height-base;
	text-align: left;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: $text-color-base;
}

.c-filter-menu-item__icon {
	color: $ui-color-success-dark;
	opacity: 0;

	transition: opacity 0.2s ease-in-out;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-filter-menu-item--active {
	font-weight: $font-weight-ultrabold;

	background-color: $ui-color-success-light;

	.c-filter-menu-item__icon {
		opacity: 1;
	}
}
