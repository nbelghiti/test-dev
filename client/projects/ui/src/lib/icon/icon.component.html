<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
