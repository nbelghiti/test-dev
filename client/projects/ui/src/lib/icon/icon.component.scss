// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
