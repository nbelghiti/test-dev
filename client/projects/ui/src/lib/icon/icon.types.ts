export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
