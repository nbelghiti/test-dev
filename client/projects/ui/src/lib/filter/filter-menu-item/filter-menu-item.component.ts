import { Component, Input } from '@angular/core';

@Component({
	selector: 'ui-filter-menu-item',
	templateUrl: './filter-menu-item.component.html',
	styleUrls: ['./filter-menu-item.component.scss']
})

export class FilterMenuItemComponent {
	@Input() public label: string = 'Label';
	@Input() public active: boolean;
	@Input() public link: any;
	@Input() public	queryParams: { [key: string]: string };
	@Input() public clickHandler: () => void = () => {};
}
