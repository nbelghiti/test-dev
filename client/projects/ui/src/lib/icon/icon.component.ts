import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
