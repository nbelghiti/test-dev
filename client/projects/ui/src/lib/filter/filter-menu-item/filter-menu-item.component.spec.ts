import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../../icon/icon.component';

import { FilterMenuItemComponent } from './filter-menu-item.component';

describe('FilterMenuItemComponent', () => {
	let component: FilterMenuItemComponent;
	let fixture: ComponentFixture<FilterMenuItemComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
			],
			declarations: [
				FilterMenuItemComponent,
				IconComponent,
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(FilterMenuItemComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
