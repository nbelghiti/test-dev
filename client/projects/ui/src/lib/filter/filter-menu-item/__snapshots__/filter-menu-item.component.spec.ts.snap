// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenuItemComponent Should render correctly 1`] = `
<ui-filter-menu-item
  clickHandler={[Function Function]}
  label={[Function String]}
>
  <button
    class="c-filter-menu-item"
    ng-reflect-klass="c-filter-menu-item"
    ng-reflect-ng-class="[object Object]"
    title="Label"
  >
    <span
      class="c-filter-menu-item__value"
    >
      Label
    </span>
    <div
      class="c-filter-menu-item__icon"
    >
      <ui-icon
        class="c-icon"
        name="check"
        ng-reflect-name="check"
      >
        <i
          class="c-icon__glyph   fas fa-check"
        />
      </ui-icon>
    </div>
  </button>
</ui-filter-menu-item>
`;
